src/vm/database/sqlite.rs

```diff
@@ -0,0 +1,305 @@
+use rusqlite::{Connection, OptionalExtension, NO_PARAMS, Row};
+use rusqlite::types::ToSql;
+
+use vm::contracts::Contract;
+use vm::database::{ContractDatabase};
+use vm::errors::{Error, ErrType, InterpreterResult as Result, IncomparableError};
+use vm::types::{Value, TypeSignature, TupleTypeSignature, AtomTypeIdentifier};
+
+
+pub struct SqliteContractDatabase {
+    conn: Option<Connection>,
+    save_point: u16
+}
+
+pub struct SqliteDataMap {
+    map_identifier: i64,
+    key_type: TypeSignature,
+    value_type: TypeSignature
+}
+
+impl SqliteContractDatabase {
+    pub fn initialize(filename: &str) -> Result<SqliteContractDatabase> {
+        let mut contract_db = SqliteContractDatabase::inner_open(filename)?;
+        contract_db.execute("CREATE TABLE IF NOT EXISTS maps_table
+                      (map_identifier INTEGER PRIMARY KEY AUTOINCREMENT,
+                       contract_name TEXT,
+                       map_name TEXT,
+                       key_type TEXT,
+                       value_type TEXT)",
+                            NO_PARAMS)?;
+        contract_db.execute("CREATE TABLE IF NOT EXISTS data_table
+                      (data_identifier INTEGER PRIMARY KEY AUTOINCREMENT,
+                       map_identifier INTEGER,
```

Comment:
Highly recommend using `FOREIGN KEY .. REFERENCES ...` here, so we don't accidentally get dangling data.